redash/query_runner/mongodb.py

```diff
@@ -130,11 +130,24 @@ class MongoDB(BaseQueryRunner):
     @classmethod
     def configuration_schema(cls):
         return {
-            "type": "object",
-            "properties": {
-                "connectionString": {"type": "string", "title": "Connection String"},
-                "dbName": {"type": "string", "title": "Database Name"},
-                "replicaSetName": {"type": "string", "title": "Replica Set Name"},
+            'type': 'object',
+            'properties': {
+                'connectionString': {
+                    'type': 'string',
+                    'title': 'Connection String'
+                },
+                'dbName': {
+                    'type': 'string',
+                    'title': "Database Name"
+                },
+                'replicaSetName': {
+                    'type': 'string',
+                    'title': 'Replica Set Name'
+                },
+                'readPreference': {
+                    'type': 'string',
+                    'title': 'Read preference (when replica set exists): primary, primaryPreferred, secondary, secondaryPreferred, or nearest'
+                }
```

Comment:
You can now specify an enum (that will create a dropdown in the UI):

```
                "readPreference": {
                    "type": "string",
                    "extendedEnum": [
                        {"value": "primary", "name": "Primary"},
                        {"value": "primaryPreferred", "name": "Primary Preferred"},
                        {"value": "secondary", "name": "Secondary"},
                        {"value": "secondaryPreferred", "name": "Secondary Preferred"},
                        {"value": "nearest", "name": "Nearest"},
                    ],
                    "title": "Replica Set Read Preference",
                },
```

Reply:
Ooo... nice... It's been a while since I've contributed (and this PR was sitting around for far too long :-) ).

Reply:
Changed to the enum + made sure readPreference is being passed to MongoClient.

I set the default as "primaryPreferred" so that it doesn't change the original logic too much. If someone will set a different value than that would be used.

Reply:
:+1: thanks! This feature (enums in configuration schema) is what delayed this PR... it just took way longer than I expected to have this feature. Then by the time we had it, this PR was out of sync.